### DISCUSSION TITLE
Flap inputs in state-space model not working

### DISCUSSION THREAD

**Discussion by PiH2000** [ImperialCollegeLondon/sharpy#190] [Q&A] [answered]
Hello,

I am trying to extract a state-space model for a wing with a flap at the tip.
However, the 2 columns in the B and D matrices corresponding to the flap inputs (deflection & velocity), and only them, are full of zeros. I have tried many different things, such as changing the surface from static to dynamic, increasing its size etc. but I keep getting the same issue.

The code that I am using is based on the FlyingWing class template. I have the attached the solver settings file. I am using a M=5, N=52 discretisation and the assembled aeroelastic system, in modal coordinates, is of the form:

Aeroelastic system assembled:
Aerodynamic states: 5460
Structural states: 10
Total states: 5470
Inputs: 17
Outputs: 15

Is there any setting in particular that would allow the flap to be modelled in the state-space system? Am I missing something?

Thank you for your help.
Pierre

[settings_M5N52u20.solver.txt](https://github.com/ImperialCollegeLondon/sharpy/files/8565257/settings_M5N52u20.solver.txt)

**Comment by ngoiz** [comment]
Hi Pierre,

Do you mind trying this in the [dev_linear](https://github.com/ImperialCollegeLondon/sharpy/tree/dev_linear) branch and seeing if this is still the case? If it is, would you mind sending me a script to generate the input files? It is not the first time we encounter this and might need to have a more in-depth look.

Thanks, 
Norberto

**Reply by PiH2000** [reply]
Hi Norberto,

Thank you for your reply. 
Unfortunately, I still get the same problem when I use the dev_linear branch instead. I'll email you the script I am using.

On a side note about the dev_linear branch, I had to comment out the 'from control import forced_response, TransferFunction' line as it produced the following error:

File "/home/pierre/sharpy-dev_linear/sharpy/generators/floatingforces.py", line 7, in <module>
from control import forced_response, TransferFunction
ModuleNotFoundError: No module named 'control'

Also, the summary outputted to the terminal describing the order of the inputs, states, and outputs in the state-space system is really useful, thank you for adding it in!

Thanks,
Pierre

**Comment by ngoiz** [comment, marked as answer]
For reference to close this discussion: it was indeed a bug and detailed in #192 and will be fixed in the next SHARPy release
